tests/test_auth.py

```diff
@@ -19,54 +19,71 @@ def setUp(self):
         self.login_route = 'api/v1/auth/login'
 
     def test_registration_conflict(self):
+        """ test registration conflict """
         req = self.client().post(self.register_route, data=self.user) # pylint: disable=unused-variable
         req1 = self.client().post(self.register_route, data=self.user) # pylint: disable=unused-variable
         self.assertEqual(req1.status_code, 409)
+        self.assertIn('exists', str(req1.data))
+        
     
     def test_registration_success(self):
+        """ test registration success """
         self.register = {"username":"spinderman", "password":"spinderman", "email":"spinderman@gmail.com"}
         req = self.client().post(self.register_route, data=self.register) # pylint: disable=unused-variable
         self.assertEqual(req.status_code, 201)
+        self.assertIn('success', str(req.data))
 
     def test_login_success(self):
+        """ test login success """
         self.user_data = {"email":"thor@gmail.com", "password":"thor"}
         req = self.client().post(self.login_route, data=self.user_data) # pylint: disable=unused-variable
         self.assertEqual(req.status_code, 200)
+        self.assertIn('token', str(req.data))
 
     def test_login_while_not_registered(self):
+        """ test not registered login """
         self.user_data = {"email":"fakethor@gmail.com", "password":"thor"}
         req = self.client().post(self.login_route, data=self.user_data) # pylint: disable=unused-variable
         self.assertEqual(req.status_code, 404)
+        self.assertIn('check your details', str(req.data))
     
-    def test_invalid_input(self):
-        self.user_data_empty = {"email":""}
+    def test_invalid_username(self):
+        """ test for invalid username """
         self.user_data_invalid_username = {"username":"#$superman", "email":"superman@gmail.com", "password":"69mansuper"}
+        req3 = self.client().post(self.register_route, data=self.user_data_invalid_username)
+        self.assertEqual(req3.status_code, 401)
+        self.assertIn('Invalid username', str(req3.data))
+
+    def test_invalid_password(self):
```

Comment:
Break down this test into smaller test suites, each testing just one single thing.

Reply:
Alright. Noted with thanks